packages/core/src/command/hl7-notification/hl7-notification-webhook-sender-direct.ts

```diff
@@ -113,8 +113,8 @@ export class Hl7NotificationWebhookSenderDirect implements Hl7NotificationWebhoo
         id: encounterId,
         cxId,
         patientId,
-        class: encounterClass.display,
-        facilityName,
+        class: encounterClass.display ?? "",
+        facilityName: facilityName ?? "",
```

Comment:
This is the actual bugfix - we throw zod errors on undefined facility name, so pass in an empty string. 

Is it better to change the zod validation to allow undefined than to fix the caller? My belief was the caller should be forced to be intentional around "not" passing in a value - we shouldn't logically accept an undefined facility. If its a data quality issue, and its empty string then its an empty string, but its not okay to just "not have" a facility. 

Don't know if that's a little ridiculous though.

Reply:
I avoid empty string like the plague. If the facility name is required, then we should fail here. Or we accept that it might not be provided in the TcmEncounter, then we'd have code to actually work with that. Currently, everyone that reads TcmEncounter expects it to be set, only to have it empty string sometimes.

Reply:
Made a change to have the endpoint accept undefined (but transform back to empty string). 

Left message in slack about why I think that's good enough for now.

Discussed in slack - [link](https://metriport.slack.com/archives/C04CXLSAF7V/p1752627981674659?thread_ts=1752622444.169279&cid=C04CXLSAF7V)